cmd/ci-operator/main.go

```diff
@@ -178,12 +179,21 @@ func main() {
 	})
 
 	if err := opt.Complete(); err != nil {
+		err = results.ForReason(results.ReasonLoadingArgs).ForError(err)
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 		opt.writeFailingJUnit(err)
 		os.Exit(1)
 	}
 
 	if err := opt.Run(); err != nil {
+		// TODO: maybe we can think of a better way to do this,
+		// but it would be awesome to not need to make the results
+		// library aware of this typed error
+		baseErr := err
+		if wrapped, ok := err.(errWroteJUnit); ok {
+			baseErr = wrapped.error
+		}
+		err = results.DefaultReason(baseErr)
```

Comment:
This causes rudundant junit reporting to test grid and all our infrastructure - if you see errWroteJunit you don't write a second junit.  I believe you wanted to set a bool and then skip line 188 unless you deliberately wanted to double write junits.

Reply:
![image](https://user-images.githubusercontent.com/1163175/80532147-ae79a300-8969-11ea-8ddb-8c5697ff2fb5.png)

I can't tell whether this PR intended to always double report the junit.  If you did, we now have to hide the duplication in more places - was there a reason not to just use the existing one?

Reply:
As discussed on chat now in #735